### ISSUE TITLE
Moq1203: false positive when Setup call is wrapped in parentheses

### ISSUE THREAD

**Issue by rjmurillo** [rjmurillo/moq.analyzers#887]
## Description

The chain walk in `MethodSetupShouldSpecifyReturnValueAnalyzer.HasReturnValueSpecification` only advances when the immediate parent of the current syntax node is a `MemberAccessExpressionSyntax`. If the `Setup(...)` invocation is wrapped in parentheses, a cast, or a conditional access expression, the walk stops early and misses the subsequent `.Returns()` call, triggering a false positive.

## Reproduction

```csharp
// This valid pattern triggers Moq1203 incorrectly:
(new Mock<IFoo>().Setup(x => x.GetValue())).Returns(42);
```

## Expected Behavior

No diagnostic. The `.Returns(42)` call satisfies the return value specification requirement.

## Actual Behavior

Moq1203 fires because the chain walk exits when it encounters `ParenthesizedExpressionSyntax` as the parent instead of `MemberAccessExpressionSyntax`.

## Proposed Fix

Skip wrapper nodes (`ParenthesizedExpressionSyntax`, `CastExpressionSyntax`, `ConditionalAccessExpressionSyntax`) before checking for `MemberAccessExpressionSyntax` in the while loop.

## Priority

Low. Parenthesizing a fluent Setup call is extremely rare in practice. Identified during review of #886.
